src/logger.js

```diff
@@ -62,167 +59,98 @@ const serializers = {
-    this.out = base;
-    this.wskLog = options.wskLog;
+function addLogger(parent, name, createFN, cfg, params) {
+  const logger = createFN(cfg, params);
+  if (!logger) {
+    return false;
   }
-
-  write(rec) {
-    const {
-      hostname,
-      level,
-      time,
-      req,
-      res,
-      ow: {
-        activationId: id = 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx',
-        actionName: name = 'unknown',
-      },
-    } = rec;
-    let {
-      msg,
-    } = rec;
-    if (this.wskLog) {
-      const line = `${LEVELS[level]} - ${msg}\n`;
-      this.out.write(line);
-    } else {
-      // use _send directly, so we can avoid the json serialization of 'rec'
-      const date = new Date(time).toJSON();
-      if (req || res) {
-        msg = JSON.stringify({
-          req,
-          res,
-        }, safeCycles(), 2);
-      }
-      const line = `<${8 + SYSLOG_LEVELS[level]}>${date} ${hostname} ${name}[0]:${id.substring(0, 16)} ${LEVELS[level]} ${msg}`;
-      this.out._send(line);
-    }
+  if (parent.loggers.has(name)) {
+    return true;
   }
+  parent.loggers.set(name, logger);
+  return true;
 }
 
-function addLogglyStream(logger, params) {
-  if (!(params && params.LOGGLY_SUBDOMAIN && params.LOGGLY_TOKEN)) {
+function addStream(logger, createFN, cfg, params) {
+  const stream = createFN(cfg, params);
+  if (!stream) {
     return false;
   }
-
-  // check if not already added
-  if (logger.streams.find((s) => s.name === 'LogglyStream')) {
-    // eslint-disable-next-line no-console
-    console.log('(Loggly stream already added)');
+  if (logger.streams.find((s) => s.name === stream.name)) {
     return true;
   }
-
-  const logglyConfig = {
-    token: params.LOGGLY_TOKEN,
-    subdomain: params.LOGGLY_SUBDOMAIN,
-  };
-  const bufferLength = 1000;
-  const bufferTimeout = 500;
-  logger.addStream({
-    type: 'raw',
-    level: config.LOG_LEVEL,
-    stream: new Bunyan2Loggly(logglyConfig, bufferLength, bufferTimeout),
-  });
+  logger.addStream(stream);
   return true;
 }
 
-function addPaperTrailStream(logger, params) {
-  if (!(params && params.PAPERTRAIL_HOST && params.PAPERTRAIL_PORT)) {
-    return false;
+/**
+ * Initializes a helix-log logger for use with an openwhisk action.
+ * It ensures that, the given logger has a default console logger configured. It also looks for
+ * credential params and tries to add additional external logger (eg. coralogix).
+ *
+ * @param {*} params                        - the openwhisk action params
+ * @param {MultiLogger} [logger=rootLogger] - a helix multi logger. defaults to the helix
+ *                                            `rootLogger`.
+ */
+function setupHelixLogger(params, logger = rootLogger) {
+  // replace default logger with our own console logger
+  if (!logger.loggers.has('openwhisk')) {
+    logger.loggers.delete('default');
+    logger.loggers.set('openwhisk', new ConsoleLogger({
+      level: 'info',
+      formatter: (msg, opts) => messageFormatSimple([msg[0]], opts),
```

Comment:
Could we please use a more robust check here?
If we ever use other helix libs or use helix-log directly this will result in a bug "log messages are being truncated".

Reply:
Thinking about it, I think we both agree that a lot of logging will actually actively use the fields feature? Especially bunyan based logging? So I think this might actually truncate themost important info?

Reply:
there is still a discrepancy between additional message arguments and objects passed to the json loggers. eg:

```
info('the result is ', result)
```
vs
```
info('completed.', { request-id: 42 })
```

in the later case, i'm not interrested in displaying the `request-id` in the console log.
(using a distinct data and message separation would help here)

Reply:
added better check.